remove pixels and demo from fish builtins. make it harder to shoot self in foot

mwookawa opened the pull request:
onesself in the foot with the terminal

jingmeihu committed:
change syscall functions

jingmeihu committed:
crystal-merge
Merge branch 'master' of github.com:Harvard-PRINCESS/barrelfish

jingmeihu committed:
update

jingmeihu committed:
make arch-specific syscall functions more generic (x86_64 and arm)

jingmeihu committed:
Merge branch 'dev' into crystal

mwookawa committed:
remove pixels and demo from fish builtins. make it harder to shoot
onesself in the foot with the terminal

mwookawa committed:
first add of lmbench3 sources

Commit:
Merge pull request #63 from Harvard-PRINCESS/dev_integrate_lmbench_as_regressions

Dev integrate lmbench as regressions

mwookawa committed:
Revert "Dev integrate lmbench as regressions"

mwookawa committed:
Merge pull request #65 from Harvard-PRINCESS/revert-63-dev_integrate_lmbench_as_regressions

Revert "Dev integrate lmbench as regressions"

penlu commented:
* this also cleans up a couple of things in the shell
* builds+boots on armv7/FVP and x86_64/QEMU, fish runs (as well as it ever has for me)
note that sometimes "execution successful" messages come back a little out of order:
>> oncore 0 /x86_64/sbin/examples/xmpl-hello
/x86_64/sbin/examples/xmpl-hello: file not found: The given name does not exist
>> oncore 0 /x86_64/sbin/examples/xmpl-helloexecution successful with domain id: (null)
/x86_64/sbin/examples/xmpl-hello: file not found: The given name does not exist
>>  
seems essentially cosmetic to me?

jingmeihu commented:
build and boot are fine, but when I run hello world example, there are some page faults. (both qemu and pandaboard have the same problem).

    oncore 0 /x86_64/sbin/examples/xmpl-hello
    spawnd.0.0: spawning /x86_64/sbin/examples/xmpl-hello on core 0
    kernel 0: user page fault in 'fish': addr b IP 5f2a4a SP 980330 error 0x4
    Can't deliver exception to thread: handler not set
    fish: unhandled page fault (error code 0x4) on b at IP 5f2a4a
    kernel_dump_ptables

jingmeihu committed:
remove arch_id syscall

jingmeihu committed:
remove arch_id syscall
Merge branch 'crystal' of github.com:Harvard-PRINCESS/Guppy into crystal

mwookawa commented:
spawn doesn't seem to pass back a correct domain_id. this was a latent bug in the barrelfish trunk. i'll take a quick look at the spawn function to see if it's easy to trace this back to domain id assignment (or if it's deprecated or ??), and if it's not, i'll use the same workaround.

Commit:
add some runtime assertions on domain ids returned from execute_program
and spawn

marikgoldstein committed:
Merge pull request #58 from Harvard-PRINCESS/crystal

make arch-specific syscall functions more generic (x86_64 and arm)

alexpatel committed:
add crontab for nightly builds

alexpatel committed:
merge GOAT nightly features into Guppy nightly build script

Features added:
- locale flags and path
- clones a fresh tree every time
- two build logs - full and summary - full is kept on build host,
summary is sent out
- reports timing for how long things took to build

Still doesn't do:
- auto updating (won't fix, if you change the nightly script stick it on
nomnomnom)
- archive build logs in own repo

mwookawa committed:
add some sanity checks and domain id debug output to fish and spawn
functions in libbarrelfish

Commit:
Merge pull request #69 from Harvard-PRINCESS/master

Merge 'master' into 'dev'

alexpatel committed:
fix script path in crontab

alexpatel committed:
fix path typo in princess nightly crontab

alexpatel committed:
make stdout/err redirect portable in princess nightly crontab

alexpatel committed:
fix list old builds in princess nightly crontab

alexpatel committed:
Merge branch 'dev' into ahp_nightly_build

alexpatel committed:
princess nightly whitespace nits

alexpatel committed:
Merge pull request #70 from Harvard-PRINCESS/ahp_nightly_build

Nightly build 2.0

mwookawa committed:
Merge branch 'master' of github.com:Harvard-PRINCESS/Guppy into dev_remove_fish_unsupported_features

mwookawa committed:
Merge branch 'dev' of github.com:Harvard-PRINCESS/Guppy into dev_remove_fish_unsupported_features

alexpatel committed:
add list of QEMU X86_64 tests that pass on trunk

alexpatel committed:
add harness script to run tests from list

alexpatel committed:
add harness/runtests.py documentation to harness README

alexpatel committed:
add Dockerfile to build and run QEMU X86_64 tests

alexpatel committed:
add docker devel/testing documentation to harness README

alexpatel committed:
update runtest.py documentation

alexpatel committed:
add initial circle CI configuration

alexpatel committed:
Merge remote-tracking branch 'origin/ahp_testing' into ahp_testing

alexpatel committed:
fix test results directory in circle CI / docker image

alexpatel committed:
add hake step to docker image config

alexpatel committed:
use harness --verbose flag in tools/harness/runtests.py

alexpatel committed:
permit testlists to have tests that pass but don't get run

- use 'R' to denote tests that pass and should be run, 'P' for tests
that pass but won't get run from the testlist
- use this to create a minimal x86 testlist that doesn't take 4 hours to
run...

alexpatel committed:
update runtest.py documentation and fix whitespace

Commit:
Merge pull request #72 from Harvard-PRINCESS/ahp_testing

Testing/development utilities

mwookawa committed:
testing circleci

mwookawa committed:
Merge branch 'dev' of github.com:Harvard-PRINCESS/Guppy into dev_remove_fish_unsupported_features
